### PR TITLE
Options use

### PR DESCRIPTION
**Pull request by azakordonets**
Not sure why, but only after exporting server var in index.js, i managed to run all the tests. I have updated travis file, so tests would run there and used my tests instead of general one you wrote, since my tests check that options are used and actually passed to library. Let me know what you think. 

Thanks.

### CODE REVIEW

**Comment by azakordonets** [comment]
I had an older version of node where everything worked fine. Now, i've updated to 5.5.0 and it's broken. Investigating why is that - sorry, travis build didn't run for some time and i rushed this a bit. Will try to fix it ASAP

**Comment by azakordonets** [comment]
Ok, i managed to fix tests on the latest version of node and it should work fine now. Let me know what you think.

File: .travis.yml [hunk 1]
```diff
@@ -1,5 +1,9 @@
 language: node_js
 
+scripts: {
+  test: "./node_modules/mocha/bin/mocha"
```
Comment (mikefrancis) [inline .travis.yml]: Why is this needed?

File: src/index.js [hunk 2]
```diff
@@ -6,7 +6,7 @@ var cowsay = require('cowsay');
 
 app.use(bodyParser.json());
 app.use(bodyParser.urlencoded({ extended: true }));
-app.listen(process.env.PORT || 3000);
+var server = app.listen(process.env.PORT || 3000);
```
Comment (mikefrancis) [inline src/index.js]: Not sure why you've changed this?

File: test/index.js [hunk 3]
```diff
@@ -1,29 +0,0 @@
-process.env.SLACK_TOKEN = 'someToken';
```
Comment (mikefrancis) [inline test/index.js]: Why have you deleted this file? You need to merge your tests into this set of stable tests, rather than just deleting the file and adding your own.

**Comment by mikefrancis** [comment]
Please see my comments, you need to take the tests I've written and add to it. You can't just remove this file and expect me to merge your tests - they need some serious work.

File: .travis.yml (hunk 1 above)
Comment (azakordonets) [inline .travis.yml]: To make travis run tests. Or you mean that without this line travis will run tests as well ?

File: src/index.js (hunk 2 above)
Comment (azakordonets) [inline src/index.js]: At first tests didn't want to successed without it. Now it's fixed

File: test/index.js (hunk 3 above)
Comment (azakordonets) [inline test/index.js]: I replaced those tests because they were only asserting for response code, but not for the response itself. From tests perspective this tests don't check whether we send correct response, or not. In this case, all the tests that i wrote were checking for all scenarios and responses correctness, so i thought it would be easier to just replace them, since my tests do the same + response body verification. But, i can merge tests as well, no problem.

**Comment by azakordonets** [comment]
Done. Merged tests

File: .travis.yml (hunk 1 above)
Comment (mikefrancis) [inline .travis.yml]: Travis will run whatever is in package.json's `scripts.test`, so this isn't needed. Please remove.

File: src/index.js [hunk 4]
```diff
@@ -19,12 +19,11 @@ app.post('/', function (req, res) {
 			.send({ text: "No text provided" });
 	}
 
-	var responseText = '```' + cowsay.say({ text: req.body.text }) + '```';
+	var eyes = req.body.e || req.body.eyes;
```
Comment (mikefrancis) [inline src/index.js]: How does this work from Slack? You'll need to edit the README too so that people don't have to guess.

File: test/index.js [hunk 5]
```diff
@@ -3,27 +3,59 @@ process.env.SLACK_TOKEN = 'someToken';
 var app = require('./../src/index');
 var request = require('supertest')(app);
 
+String.prototype.format = function (placeholders) {
```
Comment (mikefrancis) [inline test/index.js]: Not keen on any of this - it's too messy. Please remove it and just make sure that we're checking for a 200 response.

**Comment by mikefrancis** [comment]
Still not confident merging this yet I'm afraid. Can you please make sure you're running your code through [JSCS](http://jscs.info/) (the `airbnb` preset) to catch any formatting issues.

File: test/index.js (hunk 5 above)
Comment (azakordonets) [inline test/index.js]: Sure, just want to understand something. If someone touches the code where params are set, how tests that check only codes will catch this ? I mean, even before i added functionality for passing this two params, i could write a test where in request body i could add "eyes", "T", params and server would return 200 response. But in response body this parameters would be displayed. So, why it would be enough to check only for status code ?

**Comment by azakordonets** [comment]
There's one more nasty thing i just discovered. It looks like i can't add additional parameters to post body from slack. So, it means that i have to put everything into "text" field and parse it. There's two options for this : 

1. Change the way we send data from slack - move from POST to GET and then parse GET url( which i guess is easier because there are libraries, that can do that for you)
2. Agree on some kind of text format and then create regexp that will parse "text" field and assign corresponding values to "eyes" and "tongue". Since we have two ways to pass this params (for "eyes" it's "e" and "eyes, and for tongue it's "T" and "tongue"), i guess we should make some special symbols in front of this key words, or put them into brackets. 

What do you think? Or i'm wrong and there's actually a way to pass this params in post body ?

File: test/index.js (hunk 5 above)
Comment (mikefrancis) [inline test/index.js]: I think until we have found a solution to sending the options through Slack that this test isn't covering the right thing anyway so let's just leave this test for now.

**Comment by mikefrancis** [comment]
Ok, I thought this might be the case. I don't think we can do option 1 as this is always a POST from Slack. I think the best thing to do is option 2 and split the `text` into params.

I would maybe do a bit of research into how other apps do this, so that we can make sure we get it right 1st time.

**Comment by azakordonets** [comment]
Sounds great. I already twitted to SlackHQ and SlackAPI teams, but no reply yet.

**Comment by mikefrancis** [comment]
Cool, thanks for doing that :smile:

**Comment by azakordonets** [comment]
From SlackAPI team - "The `text` field contains all the text following the `/command` — you may need to parse this for multiple args." So, yeah, let's agree on some text format . How about : 

/moo hello this is a text [eyes **] [tongue U]
/moo hello this is a text [e **] [T U] 

What do you think ?

**Comment by mikefrancis** [comment]
Sounds good!
On Mon, 25 Jan 2016 at 6:10 pm, Andrew Zakordonets <notifications@github.com>
wrote:

> From SlackAPI team - "The text field contains all the text following the
> /command — you may need to parse this for multiple args." So, yeah, let's
> agree on some text format . How about :
>
> /moo hello this is a text [eyes **] [tongue U]
> /moo hello this is a text [e **] [T U]
>
> What do you think ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mikefrancis/moo/pull/10#issuecomment-174606698>.
>
-- 
_____________

mikefrancis.me

@_mikefrancis

**Comment by azakordonets** [comment]
Ok, then i'll adjust code then.

**Comment by azakordonets** [comment]
Ok, finished. I went into an easier way and i don't into account scenario where arguments are passed inside the string. So, in case some one want to write something like : 
```
/moo Hi, i have blue [eyes **] and red [tongue U]
```

Then it will show in text field only `Hi, i have blue` and then a moo with `**` eyes and `U` tongue. I also decided to keep it simple, so there's only one way to specify eyes and tongue params - `eyes` and `tongue` keys. Let me know what you think.

**Comment by mikefrancis** [comment]
Looks good to me. Thanks for all your hard work! Merged.

**Comment by azakordonets** [comment]
Thanks for letting me learn how to code javascript apps :)
